EOS - Plugin Disabling Quick Icon Display and adding correct text definitions

mfleeson opened the pull request:
Disabling Quick Icon Display and adding correct text definitions and Joomla 5 links

Pull Request for Issue # .

### Summary of Changes
Changed announcement texts to approved wording
Changed links to Joomla 5
Commented out QuickIcon display code


### Testing Instructions
Edit Line 42 'private const EOS_DATE = '2025-10-17'; and change 2025 to 2023


### Actual result BEFORE applying this Pull Request

As well as the banner appearing a quickicon in the box near Joomla Update would appear

### Expected result AFTER applying this Pull Request

The quickicon no longer appears

### Link to documentations
Please select:
- [ ] Documentation link for docs.joomla.org: <link>
- [ ] No documentation changes for docs.joomla.org needed

- [ ] Pull Request link for manual.joomla.org: <link>
- [ ] No documentation changes for manual.joomla.org needed

mfleeson committed:
Disabling Quick Icon Display and adding correct text definitions and Joomla 5 links

Disabling Quick Icon Display and adding correct text definitions and Joomla 5 links

zero-24 reviewed (dismissed):
Hi please add back the `Special:MyLanguage` to the docs URLs. This allows docs.joomla.org to redirect to an translated page when that is avaible.

mfleeson committed:
Adding Special:MyLanguage to messageLinks

zero-24 commented:
Thanks

laoneo commented on plugins/quickicon/eos/src/Extension/Eos.php:
```diff
@@ -163,12 +163,12 @@ private function saveParams(): bool
-            ->where($db->quoteName('folder') . ' = ' . $db->quote('quickicon'))
-            ->where($db->quoteName('element') . ' = ' . $db->quote('eos'))
-            ->bind(':params', $params);
+                  ->update($db->quoteName('#__extensions'))
```
Please revert that change, we intend only one tab.

mfleeson replied:
Done

laoneo commented on plugins/quickicon/eos/src/Extension/Eos.php:
```diff
@@ -135,9 +135,9 @@ public function getEndOfServiceNotification(QuickIconsEvent $event): void
+        // The message as quickicon - The next section is commented out to stop the display of the quickicon. Uncomment to re-enable.
 
-        $result[] = [
+        /*$result[] = [
```
Please remove all the commented code, we have a history in git, so we can always revert to the old state.

mfleeson replied:
Done

laoneo commented on administrator/language/en-GB/plg_quickicon_eos.ini:
```diff
@@ -3,37 +3,19 @@
 ; Note : All ini files need to be saved as UTF-8
 
-PLG_QUICKICON_EOS="Quick Icon - Joomla! End Of Support Notification"
+PLG_QUICKICON_EOS="Quick Icon - Joomla 4.4 End Of Support Notification"
```
Should the work Joomla not have an ! at the end?

mfleeson replied:
The string was supplied by Jen Gress reporting back from the team deciding on the wording for this. I'll check.

mfleeson replied:
Checked with Jen and documention. Changed all references of Joomla to Joomla! in the language strings

brianteeman replied:
Please see the style guide for this https://manual.joomla.org/docs/user-interface-text/joomla_name_usage

mfleeson replied:
Thanks Brian, I will remove the ! after the Joomla name in the language strings.
Relevent section in Brian's link: Rule: No Bang!
References to Joomla in language strings should not include the exclamation point unless it is at the end of a sentence that is intended to be exclamatory.

mfleeson replied:
Fixed

mfleeson committed:
Updating Joomla x.x to Joomla! x.x in language definitions and formatting

Changing language definitions to use Joomla!
Removing commented out code
Fixing indent.

mfleeson committed:
Removing ! from Language strings for Joomla!

Based on doc rules:
Rule: No Bang!
References to Joomla in language strings should not include the exclamation point unless it is at the end of a sentence that is intended to be exclamatory.

laoneo committed:
Merge branch '4.4-dev' into End-Of-Support-Plugin-Update

sandewt commented:
Why this icon twice? That makes it messy.

![issue-40774](https://github.com/joomla/joomla-cms/assets/9271775/b16a95f3-ba58-497f-8c9f-045151d52e16)

brianteeman commented on administrator/language/en-GB/plg_quickicon_eos.ini:
```diff
@@ -3,37 +3,19 @@
-PLG_QUICKICON_EOS_MESSAGE_INFO_01="<p>The latest version of Joomla has arrived! Find out all that it has to offer you. Check the landing page for <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">its features</a> and improvements.</p>"
-PLG_QUICKICON_EOS_MESSAGE_INFO_01_SHORT="The latest version of Joomla has arrived! <span class=\"icon-new-tab\"></span>"
-PLG_QUICKICON_EOS_MESSAGE_INFO_02="<p>When is the time to migrate to the next version? Once the extensions your site needs are compatible. Learn <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">how to use the Pre-Update Checker</a>. </p>"
+PLG_QUICKICON_EOS_MESSAGE_ERROR_SUPPORT_ENDED="<p>Support has ended for your version of Joomla 4.4. <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Upgrade to Joomla 5</a> <span class=\"icon-new-tab\"></span>as soon as possible.</p>"
```
```suggestion
PLG_QUICKICON_EOS_MESSAGE_ERROR_SUPPORT_ENDED="<p>Support has ended for your version of Joomla 4.4. <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Upgrade to Joomla 5</a> as soon as possible.</p>"
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_eos.ini:
```diff
@@ -3,37 +3,19 @@
-PLG_QUICKICON_EOS_MESSAGE_INFO_02="<p>When is the time to migrate to the next version? Once the extensions your site needs are compatible. Learn <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">how to use the Pre-Update Checker</a>. </p>"
+PLG_QUICKICON_EOS_MESSAGE_ERROR_SUPPORT_ENDED="<p>Support has ended for your version of Joomla 4.4. <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Upgrade to Joomla 5</a> <span class=\"icon-new-tab\"></span>as soon as possible.</p>"
+PLG_QUICKICON_EOS_MESSAGE_ERROR_SUPPORT_ENDED_SHORT="Support has ended for Joomla 4.4 <span class=\"icon-new-tab\"></span>"
+PLG_QUICKICON_EOS_MESSAGE_INFO_01="<p>Joomla 5 has arrived! Find out all that Joomla 5 has to offer you. Check the landing page for <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Joomla 5 features</a> <span class=\"icon-new-tab\"></span>and improvements.</p>"
```
```suggestion
PLG_QUICKICON_EOS_MESSAGE_INFO_01="<p>Joomla 5 has arrived! Find out all that Joomla 5 has to offer you. Check the landing page for <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Joomla 5 features</a> and improvements.</p>"
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_eos.ini:
```diff
@@ -3,37 +3,19 @@
+PLG_QUICKICON_EOS_MESSAGE_ERROR_SUPPORT_ENDED_SHORT="Support has ended for Joomla 4.4 <span class=\"icon-new-tab\"></span>"
+PLG_QUICKICON_EOS_MESSAGE_INFO_01="<p>Joomla 5 has arrived! Find out all that Joomla 5 has to offer you. Check the landing page for <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Joomla 5 features</a> <span class=\"icon-new-tab\"></span>and improvements.</p>"
+PLG_QUICKICON_EOS_MESSAGE_INFO_01_SHORT="Joomla 5 has arrived! <span class=\"icon-new-tab\"></span>"
+PLG_QUICKICON_EOS_MESSAGE_INFO_02="<p>When is the time to upgrade to Joomla 5? Once the extensions your site needs are compatible. Learn <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">how to use the Pre-Update Checker</a>. <span class=\"icon-new-tab\"></span></p>"
```
```suggestion
PLG_QUICKICON_EOS_MESSAGE_INFO_02="<p>When is the time to upgrade to Joomla 5? Once the extensions your site needs are compatible. Learn <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">how to use the Pre-Update Checker</a>.</p>"
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_eos.ini:
```diff
@@ -3,37 +3,19 @@
-PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY_SHORT="This version of Joomla ends support on %1$s. <span class=\"icon-new-tab\"></span>"
-PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING="<p>Support ends on %1$s for this version of Joomla. <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Migrate to the latest version of Joomla</a> as soon as possible.</p>"
-PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING_SHORT="End of support for this version of Joomla on %1$s. <span class=\"icon-new-tab\"></span>"
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY="<p>Joomla 4.4 has entered security only mode. Support ends %1$s. Start <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">planning to upgrade</a> <span class=\"icon-new-tab\"></span>to Joomla 5 today.</p>"
```
```suggestion
PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY="<p>Joomla 4.4 has entered security only mode. Support ends %1$s. Start <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">planning to upgrade</a> to Joomla 5 today.</p>"
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_eos.ini:
```diff
@@ -3,37 +3,19 @@
-PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING="<p>Support ends on %1$s for this version of Joomla. <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Migrate to the latest version of Joomla</a> as soon as possible.</p>"
-PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING_SHORT="End of support for this version of Joomla on %1$s. <span class=\"icon-new-tab\"></span>"
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY="<p>Joomla 4.4 has entered security only mode. Support ends %1$s. Start <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">planning to upgrade</a> <span class=\"icon-new-tab\"></span>to Joomla 5 today.</p>"
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY_SHORT="Joomla 4.4 to end support on %1$s. <span class=\"icon-new-tab\"></span>"
```
```suggestion
PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY_SHORT="Joomla 4.4 to end support on %1$s."
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_eos.ini:
```diff
@@ -3,37 +3,19 @@
-PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING_SHORT="End of support for this version of Joomla on %1$s. <span class=\"icon-new-tab\"></span>"
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY="<p>Joomla 4.4 has entered security only mode. Support ends %1$s. Start <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">planning to upgrade</a> <span class=\"icon-new-tab\"></span>to Joomla 5 today.</p>"
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY_SHORT="Joomla 4.4 to end support on %1$s. <span class=\"icon-new-tab\"></span>"
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING="<p>Support ends on %1$s for Joomla 4.4 <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Upgrade to Joomla 5</a> <span class=\"icon-new-tab\"></span>as soon as possible.</p>"
```
```suggestion
PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING="<p>Support ends on %1$s for Joomla 4.4 <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Upgrade to Joomla 5</a> as soon as possible.</p>"
```

brianteeman commented on administrator/language/en-GB/plg_quickicon_eos.ini:
```diff
@@ -3,37 +3,19 @@
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY="<p>Joomla 4.4 has entered security only mode. Support ends %1$s. Start <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">planning to upgrade</a> <span class=\"icon-new-tab\"></span>to Joomla 5 today.</p>"
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY_SHORT="Joomla 4.4 to end support on %1$s. <span class=\"icon-new-tab\"></span>"
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING="<p>Support ends on %1$s for Joomla 4.4 <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Upgrade to Joomla 5</a> <span class=\"icon-new-tab\"></span>as soon as possible.</p>"
+PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING_SHORT="End of support for Joomla 4.4 on %1$s. <span class=\"icon-new-tab\"></span>"
```
```suggestion
PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING_SHORT="End of support for Joomla 4.4 on %1$s."
```

mfleeson replied:
Thanks Brian, Not sure how but language definitions wound back. I have uploaded the correct versions from my test server.

sandewt commented on plugins/quickicon/eos/src/Extension/Eos.php:
```diff
@@ -214,7 +199,7 @@ private function getMessageInfo(int $monthsUntilEOS, int $inverted): array
                 'quickiconText' => 'PLG_QUICKICON_EOS_MESSAGE_ERROR_SUPPORT_ENDED_SHORT',
                 'messageType'   => 'error',
                 'image'         => 'fa fa-life-ring',
-                'messageLink'   => 'https://docs.joomla.org/Special:MyLanguage/Planning_for_Mini-Migration_-_Joomla_3.10.x_to_4.x',
+                'messageLink'   => 'https://docs.joomla.org/Special:MyLanguage/Joomla_4.4.x_to_5.x_Planning_and_Upgrade_Step_by_Step',
                 'groupText'     => 'PLG_QUICKICON_EOS_GROUPNAME_EOS',
                 'snoozable'     => false,
```
The following variables have become obsolete: quickiconText, image, groupText.
See also the code elsewhere in this file.

mfleeson replied:
These fields were in use before the requirements changed and the quickicon display code was removed. Removed now.

sandewt replied:
Don't forget to remove the associated language strings.

mfleeson committed:
Removing unnecessary field values, changing text back to correct non icon-new-tab

Not sure how but language definitions wound back. Here's the correct versions from my test server.
Also removed the extra fields that are now not needed as displaying a quickicon is no longer in the functionality requirements.

mfleeson committed:
Merge branch 'End-Of-Support-Plugin-Update' of https://github.com/mfleeson/joomla-cms into End-Of-Support-Plugin-Update

mfleeson committed:
Formatting fix

sandewt commented on plugins/quickicon/eos/src/Extension/Eos.php:
```diff
@@ -130,25 +130,6 @@ public function getEndOfServiceNotification(QuickIconsEvent $event): void
             ->registerAndUseScript('plg_quickicon_eos.script', 'plg_quickicon_eos/snooze.js', [], ['type' => 'module']);
 
         $result               = $event->getArgument('result', []);
-        $messageTextQuickIcon = sprintf(
-            $app->getLanguage()->_($this->currentMessage['quickiconText']),
-            HTMLHelper::_('date', Eos::EOS_DATE, $app->getLanguage()->_('DATE_FORMAT_LC3'))
-        );
```
Line 132-136: Is this functionality still necessary?

mfleeson replied:
That code has been removed along with line 132.

mfleeson committed:
Cleaning code

Removing unnecessary language definitions
Removing unneeded code

sandewt commented:
> Disabling Quick Icon Display ...

You removed the quickicon functionality. The plugin is still called `<name>plg_quickicon_eos</name>` and `group="quickicon"`

The naming of this plugin no longer corresponds to what the plugin shows. Is the naming still correct? Is there possibly a better alternative?

laoneo commented:
This plugin works now in a similar way as the php EOS plugin. So for now it is ok to have it in the quickicon group. On some point we need to move it to a new group.

webgras commented:
I have tested this item :white_check_mark: successfully on fda8a6cea0bb181969aef5e2ef463eab408aa9f8<br /><br />Banner works fine, tested all the dates.<hr /><sub>This comment was created with the <a href="https://github.com/joomla/jissues">J!Tracker Application</a> at <a href="https://issues.joomla.org/tracker/joomla-cms/40774">issues.joomla.org/tracker/joomla-cms/40774</a>.</sub>

MacJoom commented:
@zero-24 is the change OK?

bayareajenn commented:
Hi! Thanks for your work!
I've got a few things:

**1. in eos.php:**
line 217, 231, 245: link should be https://docs.joomla.org/Special:MyLanguage/Joomla_4.4.x_to_5.x_Planning_and_Upgrade_Step_by_Step
line 273: link should go to https://joomla.org/5
_Will you please fix those links?_

**2. in the language strings:**
line 10 PLG_QUICKICON_EOS_MESSAGE_ERROR_SUPPORT_ENDED="<p>Support has ended for your version of Joomla. <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Migrate to the next version</a> as soon as possible.</p>"
_Will you please change the word "migrate" to "upgrade"?_

line 14 PLG_QUICKICON_EOS_MESSAGE_INFO_02="<p>When is the time to migrate to the next version? Once the extensions your site needs are compatible. Learn <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">how to use the Pre-Update Checker</a>. </p>"
_Will you please change the word "migrate" to "upgrade"?_

line 16 PLG_QUICKICON_EOS_MESSAGE_WARNING_SECURITY_ONLY="<p>This version of Joomla has entered security only mode. Support ends %1$s. Start <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">planning to migrate</a> to the next version of Joomla today.</p>"
_Will you please change the word "migrate to "upgrade"?_

line 18 PLG_QUICKICON_EOS_MESSAGE_WARNING_SUPPORT_ENDING="<p>Support ends on %1$s for this version of Joomla. <a href=\"%2$s\" target=\"_blank\" rel=\"noopener noreferrer\">Migrate to the latest version of Joomla</a> as soon as possible.</p>" 
_Will you please change the word "Migrate" to "Upgrade"?_

Ok, I think that's it :) Thank YOU!!

sandewt commented:
@bayareajenn It is more effective to click on the relevant line in the code and add the suggestion.

![issue_4074](https://github.com/joomla/joomla-cms/assets/9271775/42ac109e-af27-4b4a-b13b-ec09812cbe7e)

bayareajenn commented:
> @bayareajenn It is more effective to click on the relevant line in the code and add the suggestion.
> 
> ![issue_4074](https://user-images.githubusercontent.com/9271775/248355396-42ac109e-af27-4b4a-b13b-ec09812cbe7e.png)

Yes, I'm sure it is. Sadly, I don't know how. I don't really know how to use github. I was in the code in the site. I didn't even think of going to the code in the PR. I'm not a developer. I apologise. The formatting went to poo even worse once I commented.

sandewt commented:
> Yes, I'm sure it is. Sadly, I don't know how. I don't really know how to use github......

Very understandable.

Click on the link, as indicated below with the red arrow at the top of this page.
You will then arrive at the relevant files.
And follow the instructions I indicated above.

![issue_4074-2](https://github.com/joomla/joomla-cms/assets/9271775/ed894b72-d8c4-4b1d-8ff6-67baac85acaf)

bayareajenn commented:
> > Yes, I'm sure it is. Sadly, I don't know how. I don't really know how to use github......
> 
> Very understandable.
> 
> Click on the link, as indicated below with the red arrow at the top of this page. You will then arrive at the relevant files. And follow the instructions I indicated above.
> 
> ![issue_4074-2](https://user-images.githubusercontent.com/9271775/248373762-ed894b72-d8c4-4b1d-8ff6-67baac85acaf.png)

Yes, I kinda got it and started adding them. Then realised that the nightly I was working off of is different from the files in here. So now I'm just thoroughly confused. I'm now unsure of what is correct and what isn't. So I'll sleep on it and work with the team. Thank you for teaching me @sandewt. I will get better over time.

mfleeson commented:
Hi Everyone. The version Jenn downloaded was the previous version before this PR. I have checked and these files are the correct ones with the correct language definitions and links.

MacJoom committed:
Merge branch '4.4-dev' into End-Of-Support-Plugin-Update

laoneo committed:
Merge branch '4.4-dev' into End-Of-Support-Plugin-Update
